File: Shop_Back/Models/Product.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Shop_Back.Models
{
    public class Product
    {
        public int Id { get; set; }
        [StringLength(255)]
        public string Title { get; set; }
        [StringLength(255)]
        public string ImageUrl { get; set; }
        [StringLength(255)]
        public string Description { get; set; }
        [Required]
        public double Price { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime LastUpdated { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }
    }
}

File: Shop_Front/Shop/src/app/models/Product.ts

export interface Product {
  id: string;
  title: string;
  imageUrl: string;
  description: string;
  price: string;
  userId: string;
}

export class Product implements Product {
  constructor() {
    this.id = '';
    this.title = '';
    this.imageUrl = '';
    this.price = '';
    this.description = '';
    this.userId = '';
  }
}
